### DISCUSSION TITLE
What is correct way to get new document from model class

### DISCUSSION THREAD

**Discussion by JANDA-Developers** [typegoose/typegoose#622] [Q&A] [answered]
I upgrade mongoose version to 6 and i found...
```ts
import {
    getModelForClass,
} from "@typegoose/typegoose";

export const ProductModel = getModelForClass(Product); // this give me mongoose model type from my class



const product = new Product();

const A = new ProductModel(product) // A is not DocumentType it is just Product...
const B = new ProductModel({...product}) //  B is DocumentType But why?

```
Why only B became a document instance ?

**Comment by hasezoey** [comment, marked as answer]
Could you provide what versions are using? (mongoose, typegoose, typescript)
And also please check that `@types/mongoose` is NOT installed

> `const A = new ProductModel(product) // A is not DocumentType it is just Product...`

but yes, this should not happend, but typegoose can not really change anything about it (this may be related to the performance fixed that were added in mongoose 6.0.x to try to make it faster, see https://github.com/Automattic/mongoose/issues/10349)

**Reply by JANDA-Developers** [reply]
@hasezoey  I thought this was intended behavior, but it doesn't seem to be.

 "@typegoose/typegoose": "9.1.0",
  "mongoose": "6.0.10",
  "typescript:  "3.9.7"

**Reply by hasezoey** [reply]
>  "typescript: "3.9.7"

did you already try typescript 4.1 (typescript version used by mongoose) or typescript 4.4, recommended to use with typegoose (since 9.0), see [Requirements](https://typegoose.github.io/typegoose/docs/guides/quick-start-guide#requirements)?

> I thought this was intended behavior, but it doesn't seem to be.

yes, a call of `new ProductModel()` should always return a Document, and so type `DocumentType` (or at least of type `mongoose.Document`)

also, what i said earlier about `but typegoose can not really change anything about it` might be kinda wrong, i will look later (after having confirmed versions) if this is a issue by typegoose or on the mongoose side

**Reply by JANDA-Developers** [reply]
@hasezoey Ahhh It works now :) upgrade typescript version to 4.4 fixed it.
Thanks!! so much
